include/openenclave/enclave.h

```diff
@@ -579,6 +580,100 @@ oe_result_t oe_get_seal_key_v2(
  */
 void oe_free_seal_key(uint8_t* key_buffer, uint8_t* key_info);
 
+/**
+ * Get enclave-specific key information that could be passed to
+ * oe_get_seal_key_v2() to derive a symmetric encryption key coupled to the
+ * enclave platform.
+ *
+ * @param[in] seal_policy The policy for the identity properties used to derive
+ * the seal key.
+ * @param[in] entropy Optional parameter containing additional entropy for key
+ * derivation. If \c NULL, oe_get_seal_key_info() will generate entropy on
+ * behalf of the caller.
+ * @param[in] entropy_size Must be \c 0 if \p entropy is \c NULL.
+ * @param[in] opt_flags \c 0 to use TEE defaults. If \c OE_SEAL_SGX_SPECIFIC is
+ * specified, the rest of \p opt_flags is interpreted as the \a attribute_mask
+ * for \c EGETKEY on SGX.
+ * @param[out] key_info On success this points to the enclave-specific key
+ * information which can be used to retrieve the key by passing it to
+ * oe_get_seal_key_v2(). Freed by calling oe_free_key().
```

Comment:
What's the advantage of needing two steps (get info then get key) rather than combining them into one call?

Reply:
The reason for that is there are more customizable fields in `sgx_key_request_t` than parameters of `oe_get_seal_key_info()`. 2 separate steps allows developers to further customize `key_info` in between. Of course the existing `oe_get_seal_key_by_policy()` may serve as the first step too. What's been added here are `entropy` and `attribute_mask`, which are 2 of the mostly used fields of `sgx_key_request_t`. But if you don't like it, I can take this API out but I guess some developers may find it inconvenient.

Reply:
https://github.com/openenclave/openenclave/blob/master/docs/ApiGuidelines.md says:

> All public APIs must be used in some sample code or snippet. Ideally, the sample should be one that is compiled as part of the build, and uses no experimental APIs. For rarely used public APIs, the sample might simply be a snippet in the doxygen comments. This means that, for any pull request to be merged that adds a public API, it must also add at least one sample or snippet showing how it is used.

Reply:
I can add a sample surely once everyone has agreed on the API definitions. 

I don't see any samples compiled as part of the build. What make target should I invoke? Could you point me to an example? Or is it ok for me to rewrite part of [data-sealing](../tree/master/samples/data-sealing/)?

Reply:
Yes modifying the existing data-sealing sample sounds good to me.